llvm/include/llvm/IR/RepoTicket.h

```diff
@@ -59,46 +56,94 @@ struct GONumber {
 
 /// A structure of a global object (GO) information.
 struct GOInfo {
-  /// GO's initial hash value which does not include the hash of its dependents.
+  /// The InitialDigest is the hash of an object itself (a function or global
+  /// variable) without consideration for any references to other objects.
   DigestType InitialDigest;
-  /// A set of global objects which the GO's hash depenendent on.
-  DependenciesType Dependencies;
-  /// A set of global variables which the GO's hash contributed to.
-  ContributionsType Contributions;
+  /// The Contributions of an object (X) are those objects (Y) which
+  /// transitively reference X and where a potential optimisation to either X or
+  /// any of Y may invalidate both.
+  GOVec Contributions;
+  /// The Dependencies of an object are those objects which it transitively
+  /// references but are not Contributions.
+  GOVec Dependencies;
 
   GOInfo() = default;
-  GOInfo(DigestType &&Digest, DependenciesType &&Dependencies,
-         ContributionsType &&Contributions)
-      : InitialDigest(std::move(Digest)), Dependencies(std::move(Dependencies)),
-        Contributions(std::move(Contributions)) {}
+  GOInfo(DigestType &&Digest, GOVec &&Contributions, GOVec &&Dependencies)
+      : InitialDigest(std::move(Digest)),
+        Contributions(std::move(Contributions)),
+        Dependencies(std::move(Dependencies)) {}
+
+#ifndef NDEBUG
+  void dump() const {
+    dbgs() << "GOInfo:";
+    dbgs() << "\n\tInitial Digest: " << InitialDigest.digest();
+    dbgs() << "\n\tContributions: [ ";
+    for (unsigned i = 0, e = Contributions.size(); i != e; ++i) {
```

Comment:
This might be simpler with a range based for and a "first" flag. (e.g. https://stackoverflow.com/a/27585064/292440).

Reply:
Fixed, thanks.